### PR TITLE
My Jetpack: add Search interstitial view

### PR DESCRIPTION
**Pull request by retrofox**
* Changes the product status handling, checking if it requires a paid plan first of all.
* Adds the `<SearchInterstitial />` component
* Register the `/add-search` route

#### Follow-up tasks

* Compute properly the price of Search [22702](https://github.com/Automattic/jetpack/issues/22702)
* Register/reuse a new Jetpack redirect to lead to the checkout page [22701](https://github.com/Automattic/jetpack/issues/22701)

Fixes #

#### Changes proposed in this Pull Request:

* My Jetpack: add Search interstitial view
* Intrdoduces status `NEEDS_PURCHASE`.

#### Testing instructions:

**With a site without a Search product**

* Go to My Jetpack
* Confirm you see the `Add Search` link in the product card

* Confirm you see the Search interstitial by clicking on the `Add Search` link

**The checkout link is not already implemented :-(**

* Buy the product from the jetpack dashboard
* Once you buy it, confirm you can activate/deactivate the Search product.

https://user-images.githubusercontent.com/77539/152606626-ed034401-7464-4764-b809-9b54de4c8aed.mov

### CODE REVIEW

File: projects/packages/my-jetpack/_inc/components/product-card/index.jsx [hunk 1]
```diff
@@ -110,12 +113,14 @@ const ProductCard = props => {
...
 	const isInactive = status === PRODUCT_STATUSES.INACTIVE;
-	const isAbsent = status === PRODUCT_STATUSES.ABSENT;
+	const isAbsent = status === PRODUCT_STATUSES.ABSENT || status === PRODUCT_STATUSES.NEEDS_PURCHASE;
+	const isPurchaseRequired = status === PRODUCT_STATUSES.NEEDS_PURCHASE;
 	const flagLabel = PRODUCT_STATUSES_LABELS[ status ];
 	const canDeactivate = ( isActive || isError ) && admin;
 
 	const containerClassName = classNames( styles.container, {
 		[ styles.plugin_absent ]: isAbsent,
+		[ styles[ 'is-purchase-required' ] ]: isPurchaseRequired,
```
Comment (oskosk) [inline projects/packages/my-jetpack/_inc/components/product-card/index.jsx:123]: Seems like this styles is not used or defined anywhere else

Comment (retrofox) [reply projects/packages/my-jetpack/_inc/components/product-card/index.jsx:123]: True... I want to leave a hint that we may change the UI based on it. Let me remove it.

Comment (retrofox) [reply projects/packages/my-jetpack/_inc/components/product-card/index.jsx:123]: I've applied the same styles that when the plugin is absent [here](https://github.com/Automattic/jetpack/pull/22687/commits/ceaaa26294c66a71a7c71afa501fd3df8aa4377f).

**Review by oskosk — APPROVED** [review]

**Comment by retrofox** [comment]
thanks

**Pull request merged**